lib/presets.js

```diff
@@ -305,21 +305,59 @@ module.exports = {
      * });
      */
     cargo: (options, cwd, version, callback) => {
-      const cargoFiles = path.join(cwd, 'Cargo.*');
+      const cargoToml = path.join(cwd, 'Cargo.toml');
+      const cargoLock = path.join(cwd, 'Cargo.lock');
       const cleanedVersion = semver.clean(version);
 
       if (!cleanedVersion) {
         return callback(new Error(`Invalid version: ${version}`));
       }
 
-      replace({
-        files: cargoFiles,
-        from: /(version\s*=\s*)"(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)"/g,
-        to: '$1"' + cleanedVersion + '"'
-      }, (error) => {
+      fs.readFile(cargoToml, {
+        encoding: 'utf8'
+      }, (error, contents) => {
         if (error) {
+          if (error.code === 'ENOENT') {
+            error.message = `No such file or directory: ${cargoToml}`;
+          }
+
           return callback(error);
         }
+
+        // Capture first `name = "..."` occurrence immediately after `[package]`
+        const matches = contents.match(/\[package\](?:\n|.)*?name\s*=\s*(?:"|')(.+)(?:"|')/m);
+        if (_.isNull(matches)) {
+          return callback(
+            new Error(`Package name not found in ${cargoToml}`)
+          );
+        }
+        const packageName = matches[1];
+
+        // Update first `version = "..."` occurrence immediately after `[package]`
+        replace({
+          files: cargoToml,
+          from: /(\[package\](?:\n|.)*?version\s*=\s*)("|')(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)("|')/m,
```

Comment:
Minor tweak, but the second `("|')` could/should be replaced by `\2`, as the current regex will also match `version = "1.0.0'` or `version = '1.0.0"`, whereas using `\2` will ensure that the quotes match up (i.e. it'll only match `version = "1.0.0"` or `version = '1.0.0'`).

Reply:
Excellent